### PR TITLE
[doc] Add a paragraph emphasizing that an implicit request must exist in CSRF

### PR DESCRIPTION
**Pull request by wsargent**
Fixes https://github.com/playframework/playframework/issues/3672

### CODE REVIEW

File: documentation/manual/working/scalaGuide/main/forms/ScalaCsrf.md [hunk 1]
```diff
@@ -57,6 +57,39 @@ The `Filters` class can either be in the root package, or if it has another name
 play.http.filters = "filters.MyFilters"
 ```
 
+### Using an implicit request
+
+All CSRF functionality assumes that a `RequestHeader` or a `Request` is available in implicit scope, and will not compile without one available. 
+
+#### Defining an implicit Request in Actions
```
Comment (gmethvin) [inline documentation/manual/working/scalaGuide/main/forms/ScalaCsrf.md]: I think this is good advice, but is there a more general place we should put this? Implicit requests are used in Play a lot.

Comment (wsargent) [inline documentation/manual/working/scalaGuide/main/forms/ScalaCsrf.md]: The bug was in reference to there not being an example -- more generally, I think we should rework:

* [ScalaActions](https://www.playframework.com/documentation/2.5.x/ScalaActions)
* [ScalaAsync](https://www.playframework.com/documentation/2.5.x/ScalaAsync)
* [ScalaTemplates](https://www.playframework.com/documentation/2.5.x/ScalaTemplates)
* [JavaTemplates](https://www.playframework.com/documentation/2.5.x/JavaTemplates)

**Comment by marcospereira** [comment]
Can we extract these code examples to external files?

**Comment by gmethvin** [comment]
Actually I guess if they were extracted to external files, we could easily include the same example in several of those pages that talk about implicit requests. It will be somewhat redundant to have the same example on multiple pages but it probably makes it more obvious to the reader of the documentation.

**Comment by wsargent** [comment]
The problem here is that many, many places in Play will work without an implicit request.  The examples deal explicitly with CSRF, so they can't be abstracted generally, and they're intentionally pseudocode, i.e. there is no `SomeCSRFAction`, there is no body included, and the template includes ellipsis.

```
@(...)(implicit request: Request[_])
```

So while I think there's a good point to having another issue open talking about implicit requests in more detail, I don't think that invalidates these three paragraphs of documentation, and I don't think they'd be improved by moving them out to snippets since they're intentionally more general than actual code.

**Pull request merged**